[hatstall] Add support for editing organizations and domains

mafesan opened the pull request:
(Edited) 

Now, every entry of the "Organizations" table in that view will show a list of domains for that organization and a "Edit" button. When clicking this button, a new view will be rendered
loading a form which allows to:
* Remove the organization (and its domains) from the database.
* Remove its domains individually.
* Add a new domain.

When the "Remove organization" button is clicked, a modal will reveal asking the user for confirmation. Once the "confirm" button is clicked, the form will be submitted and that organization and
its domains will be deleted from the database. Note that the modal does not pop-up if you remove domains individually.

These are some screenshots where this feature can be seen:

![Screenshot from 2019-12-23 18-07-25](https://user-images.githubusercontent.com/9032790/71370745-234d6280-25af-11ea-8cb4-586c67874ad4.png)

![Screenshot from 2019-12-23 17-39-43](https://user-images.githubusercontent.com/9032790/71370116-23e4f980-25ad-11ea-8f23-fa77bb2d70b2.png)

sduenas commented:
When the `Edit` panel is active, if you click on `Add` (to add a new organization) the panel is still active but align to the right. I would have expected that it will close the `Edit` panel when `Add` is clicked.

![imagen](https://user-images.githubusercontent.com/833352/71365755-3d337900-25a0-11ea-9d1b-671fbe4f9349.png)

mafesan commented:
Thank you for your feedback @sduenas ! I have just updated the description.

These are the changes:

* I have added the modal asking for confirmation only when trying to remove the whole organization. Note that the modal does not pop-up if you remove domains individually.
* I have changed the behavior of the "Edit organization" form after your comment. Now, if it is displayed and you click the "Add" button for the "Add organization" form, the other form will not be displayed.

sduenas commented:
LGTM
